### PR TITLE
feat: Add end to end tests for querier and schema client

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
re: https://github.com/influxdata/influxdb_iox/issues/4102

I want it to be easy to add new targeted end to end tests for things like `namespace` API and write tokens

# Changes
1. Refactor common code into `MiniCluster` fixture
1. Add a test for reading from querier
2. Add client tests for testing schema API (in addition to existing CLI test, added in https://github.com/influxdata/influxdb_iox/pull/4132)

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_ng_cases/ingester.rs [hunk 1]
```diff
@@ -60,20 +50,4 @@ async fn router2_through_ingester() {
         "+------+------+--------------------------------+-----+",
     ];
     assert_batches_sorted_eq!(&expected, &query_results);
-
-    // Validate the output of the schema CLI command
-    Command::cargo_bin("influxdb_iox")
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_ng_cases/ingester.rs:65]: moved to its own test (schema.rs)

File: influxdb_iox/tests/end_to_end_ng_cases/querier.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+use arrow_util::assert_batches_sorted_eq;
+use http::StatusCode;
+use test_helpers_end_to_end_ng::{
+    maybe_skip_integration, query_until_results, MiniCluster, TestConfig,
+};
+
+#[tokio::test]
+async fn basic_on_parquet() {
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_ng_cases/querier.rs:8]: here is the end to end test for the querier

File: influxdb_iox/tests/end_to_end_ng_cases/schema.rs [hunk 3]
```diff
@@ -0,0 +1,74 @@
+use assert_cmd::Command;
+use http::StatusCode;
+use predicates::prelude::*;
+use test_helpers_end_to_end_ng::{maybe_skip_integration, MiniCluster, TestConfig};
+
+/// Test the schema client
+#[tokio::test]
+async fn ingester_schema_client() {
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_ng_cases/schema.rs:8]: here is the new test of the schema client API

File: test_helpers_end_to_end_ng/src/mini_cluster.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
+use http::Response;
+use hyper::Body;
+
+use crate::{rand_name, write_to_router, ServerFixture, TestConfig, TestServer};
+
+/// Structure that holds NG services and helpful accessors
```
Comment (alamb) [inline test_helpers_end_to_end_ng/src/mini_cluster.rs:6]: structure to reduce boilerplate across end to end tests

**Review by crepererum — APPROVED** [review]
:tada:

**Pull request merged**